jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/LoggingMXBeanImpl.java

```diff
@@ -49,8 +57,60 @@
 /*[ENDIF]*/
 			PlatformLoggingMXBean {
 
-	private static final LoggingMXBeanImpl instance = new LoggingMXBeanImpl();
+	private static final LoggingMXBeanImpl instance;
+
+/*[IF Sidecar19-SE]
+	private static final Module java_logging;
+/*[ENDIF]*/
+	private static final Class<?> logManagerClass;
+	private static final Class<?> loggerClass;
+	private static final Class<?> levelClass;
+	private static final Method logManager_getLogManager;
+	private static final Method logManager_getLogger;
+	private static final Method logManager_getLoggerNames;
+	private static final Method logger_getLevel;
+	private static final Method logger_getParent;
+	private static final Method logger_getName;
+	private static final Method logger_setLevel;
+	private static final Method level_getName;
+	private static final Method level_parse;
+	private static final String logManager_LOGGING_MXBEAN_NAME;
+	
+	static {
+		try {
+/*[IF Sidecar19-SE]
+/*[IF Sidecar19-SE-B165]
+			java_logging = ModuleLayer.boot().findModule("java.logging").get();
+/*[ELSE]
+			java_logging = Layer.boot().findModule("java.logging").get();
+/*[ENDIF]			
+			logManagerClass = Class.forName(java_logging, "java.util.logging.LogManager");
+			loggerClass = Class.forName(java_logging, "java.util.logging.Logger");
+			levelClass = Class.forName(java_logging, "java.util.logging.Level");
+/*[ELSE]*/
+			logManagerClass = Class.forName("java.util.logging.LogManager");
+			loggerClass = Class.forName("java.util.logging.Logger");
+			levelClass = Class.forName("java.util.logging.Level");
+/*[ENDIF]*/		
+			logManager_getLogManager = logManagerClass.getMethod("getLogManager");
+			logManager_getLogger = logManagerClass.getMethod("getLogger", String.class);
+			logManager_getLoggerNames = logManagerClass.getMethod("getLoggerNames");
+			logger_getLevel = loggerClass.getMethod("getLevel");
+			logger_getParent = loggerClass.getMethod("getParent");
+			logger_getName = loggerClass.getMethod("getName");
+			logger_setLevel = loggerClass.getMethod("setLevel", levelClass);
+			level_getName = levelClass.getMethod("getName");
+			level_parse = levelClass.getMethod("parse", String.class);
+			logManager_LOGGING_MXBEAN_NAME = logManagerClass.getField("LOGGING_MXBEAN_NAME").get(null).toString();
 
+		} catch (Exception e) {
+			throw newInternalInvokeError(e);
```

Comment:
The throw isn't required, as newInternalInvokeError() doesn't ever return but throws itself. Applies to all places that use newInternalInvokeError. This also makes "newInternalInvokeError" a strange name for this helper.

Reply:
By adding the throw, it's immediately obvious that execution cannot continue past this point.  I strongly believe in keeping the `throw` there.

Reply:
Then the helper could return the Throwable instead of throwing it.

Reply:
The helper deliberately throws it to catch places were the caller has forgotten to write the throws.  It makes the code clear that the exception will be thrown and covers the case were the caller forgets the `throws`.

Its a pattern used heavily in the JSR 292 code.